sysl2/sysl/Parser_test.go

```diff
@@ -70,36 +78,71 @@ func parseComparable(filename, root string) (*sysl.Module, error) {
 	return module, nil
 }
 
-func parseAndCompare(filename, root string) (bool, error) {
-	goModule, err := parseComparable(filename, root)
+func parseAndCompare(filename, root, golden string, goldenModule *sysl.Module) (bool, error) {
+	module, err := parseComparable(filename, root)
 	if err != nil {
 		return false, err
 	}
 
-	pyModule, err := pyParse(filename, root)
+	if proto.Equal(goldenModule, module) {
+		return true, nil
+	}
+
+	generated, err := ioutil.TempFile("", "sysl-test-*.textpb")
 	if err != nil {
 		return false, err
 	}
-	if proto.Equal(pyModule, goModule) {
-		return true, nil
-	}
+	defer generated.Close()
```

Comment:
`generated.Close()` is called further down on line 105. The deferred call is generating an error but is ignored due to defer.

Reply:
Both calls are needed and it's easier to call `Close` redundantly than to check whether it has already been called. I could add a check or simply an explanatory comment.

Reply:
uh. more complicated code with defer. reminds me of c++ destructors/RAII. Goes to show error handling is not that simple. You end up having multiple code paths to think about.

@marcelocantos Is the intent of `defer close` to close the file in case of returning from  99 and 103 and the close at 103 is to make sure its closed before diff?

Reply:
Done (added a check).